homeassistant/components/melnor/switch.py

```diff
@@ -0,0 +1,90 @@
+"""Support for Melnor RainCloud sprinkler water timer."""
+
+from __future__ import annotations
+
+from typing import Any, cast
+
+from melnor_bluetooth.device import Valve
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DEFAULT_CONNECTION_TIMEOUT, DOMAIN
+from .models import MelnorBluetoothBaseEntity, MelnorDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up the switch platform."""
+    switches = []
+
+    coordinator: MelnorDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id][
+        "coordinator"
+    ]
+
+    await coordinator.data.connect(timeout=DEFAULT_CONNECTION_TIMEOUT)
+    if not coordinator.data.is_connected:
+        raise PlatformNotReady(f"Failed to connect to: {coordinator.data.mac}")
+
```

Comment:
```suggestion
```

It looks like you already do this in `async_setup_entry` so you'll know the device is already connected since `await coordinator.async_config_entry_first_refresh()` will update the data or fail and raise `ConfigEntryNotReady` so you shouldn't be able to get here

Reply:
Perfect! I wasn't totally sure how this wired together. I'll take a swing at simplifying this!